### DISCUSSION TITLE
How to populate "enum" from enum's toString method

### DISCUSSION THREAD

**Discussion by ghallas** [victools/jsonschema-generator#253] [Q&A] [answered]
Hello! I wanna start by saying "nice work!".

and now to the question: I have an enum 

```
public enum JobSchedule {
    FULL_TIME("Full time"),
    PART_TIME("Part time");

    private String stringValue;

    JobSchedule(String stringValue) {
        this.stringValue = stringValue;
    }

    @Override
    public String toString() {
        return stringValue;
    }
}
```

When I set this type to a field, what I get is:

```
"enum" : [ "FULL_TIME", "PART_TIME" ]
```

while what I would like is 

```
"enum" : [ "Full time", "Part time" ]
```

Is there a way to achieve that?

Thanx in advance

**Comment by ghallas** [comment, marked as answer]
Sloppy work on my side, it's quite apparent:

```
configBuilder.with(FLATTENED_ENUMS_FROM_TOSTRING);
```

**Reply by CarstenWickner** [reply]
Yes, exactly.
It's documented here: https://victools.github.io/jsonschema-generator/#generator-options
> `Option.FLATTENED_ENUMS_FROM_TOSTRING`
> 
> **Behavior if included**
> Treating enountered enum types as `{ "type": "string" }` schema with the `toString()` values of the enum constants being listed as its enum values.

The various out-of-the-box approaches for handling enums are also listed in the FAQs: https://victools.github.io/jsonschema-generator/#what-about-enums
